src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -270,6 +285,29 @@ std::pair<int, int> AdaptDispatch::_GetVerticalMargins(const til::rect& viewport
     return { topMargin, bottomMargin };
 }
 
+// Routine Description:
+// - Returns the coordinates of the horizontal scroll margins.
+// Arguments:
+// - bufferWidth - The width of the buffer
+// Return Value:
+// - A std::pair containing the left and right coordinates (inclusive).
+std::pair<int, int> AdaptDispatch::_GetHorizontalMargins(const til::CoordType bufferWidth) noexcept
+{
+    // If the left is out of range, reset the margins completely.
+    const auto rightmostColumn = bufferWidth - 1;
```

Comment:
will this ever be called with a width from a different text buffer, or a width that isn't the width of the active text buffer? If so, there is a risk that we blow up global state based on a local parameter

Reply:
Well there's the alt buffer, and eventually we will have additional page buffers, but they should all have the same width. And in any event, the margins are meant to be a shared global state, so if they get reset by one buffer, then it's expected that they'll be reset everywhere.

Reply:
Although, now that I think about it, there may be a case where we're passing in the line width rather than the buffer width. So on a double-width line, the margin could be out of range - and thus reset - when perhaps it shouldn't be. I'm not actually sure how that is expected to work. It might still be correct as it is.

Reply:
> So on a double-width line, the margin could be out of range

I've just been trying to trigger this now, only to realise it's not possible. I forgot that line renditions are disabled when horizontal margins are active, so this is fine as it is. Still would be nice if we could do these calculations when resizing (as I explained above), but it's not essential for now.